WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1525,18 +1520,10 @@ private void primaryAction() {
                 showUpdateConfirmationDialogAndUploadPost();
                 return;
             case PUBLISH_NOW:
-                if (AppPrefs.isAsyncPromoRequired()) {
-                    showAsyncPromoDialog(mPost.isPage(), false);
-                } else {
-                    showPublishConfirmationDialogAndPublishPost();
-                }
+                showPublishConfirmationDialogAndPublishPost();
                 return;
             case SCHEDULE:
-                if (AppPrefs.isAsyncPromoRequired()) {
-                    showAsyncPromoDialog(mPost.isPage(), true);
-                } else {
-                    uploadPost(false);
-                }
+                uploadPost(false);
```

Comment:
We can move this to the "other cases" below:

```java
            // In other cases, we'll upload the post without changing its status
            case SCHEDULE:
            case SUBMIT_FOR_REVIEW:
            case SAVE:
                uploadPost(false);
                break;
```

Reply:
Addressed in 0fafdc03907ecff2088d912ef368593e45832ce3